sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -38,15 +58,27 @@ impl Policy for BearerTokenCredentialPolicy {
         request: &mut Request,
         next: &[Arc<dyn Policy>],
     ) -> PolicyResult {
-        let scopes = self
-            .scopes
-            .iter()
-            .map(|s| s.as_str())
-            .collect::<Vec<&str>>();
-        let access_token = self.credential.get_token(&scopes).await?;
-        let token = access_token.token.secret();
+        let access_token = self.access_token.read().await;
+
+        if let Some(token) = &(*access_token) {
+            if token.is_expired(Some(DEFAULT_REFRESH_TIME)) {
+                drop(access_token);
```

Comment:
I know previously we had a comment about this being unnecessary, but I found in my testing that the lock was getting hung up (from the read lock being acquired in L61 above). I was under the impression that we are able to have multiple readers **and** only one writer, but it seems like it is actually an or according to the [docs](https://docs.rs/async-lock/latest/async_lock/struct.RwLock.html#)
```
running 1 test
test blob_client::tests::test_download_blob has been running for over 60 seconds

Reply:
It's about scope. You established the scope in L61 so it will live until the outer scope is closed. But yes, this is functionally correct.